cmd/apichecker/apicheck.go

```diff
@@ -0,0 +1,1232 @@
+	pathGetAllLists    = "https://api.trello.com/1/boards/%s/lists?cards=none&card_fields=all&filter=open&fields=all&key=%s&token=%s"
+	pathNewCard        = "https://api.trello.com/1/cards?%s&key=%s&token=%s"
+	pathChecklists     = "https://api.trello.com/1/checklists/%s/checkItems?%s&key=%s&token=%s"
+	pathChecklistItems = "https://api.trello.com/1/checklists/%s?fields=name&cards=all&card_fields=name&key=%s&token=%s"
+	pathUpdateItems    = "https://api.trello.com/1/cards/%s/checkItem/%s?%s&key=%s&token=%s"
+	trelloKey          = ""
+	trelloToken        = ""
+	trelloChecklistID  = "5dfc5a5377835d0ba025787a"
+	trelloCardID       = "5dfc54b96da13a6ac5ceca97"
+	complete           = "complete"
+	incomplete         = "incomplete"
+)
+
+// Config is a format for storing update data
+type Config struct {
+	ConfCardID      string         `json:"CardID"`
+	ConfChecklistID string         `json:"ChecklistID"`
+	ConfKey         string         `json:"Key"`
+	ConfToken       string         `json:"Token"`
+	Exchanges       []ExchangeInfo `json:"Exchanges"`
+}
+
+var (
+	verbose                                           bool
+	apiKey, apiToken, updateChecklistID, updateCardID string
+	configData, testConfigData                        Config
+)
+
+func main() {
+	flag.StringVar(&apiKey, "key", "", "its an API Key for trello")
+	flag.StringVar(&apiToken, "token", "", "its an API Token for trello")
+	flag.StringVar(&updateChecklistID, "checklistid", "5dfc5a5377835d0ba025787a", "checklist id for trello")
+	flag.StringVar(&updateCardID, "cardid", "5dfc54b96da13a6ac5ceca97", "card id for trello")
+	flag.BoolVar(&verbose, "verbose", false, "Increases logging verbosity for API Update Checker")
+	flag.Parse()
+	var err error
+	testConfigData, err = ReadFileData(testJSONFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	configData, err = ReadFileData(jsonFile)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if areAPIKeysSet() {
+		// Assumption here is that if api key n token are set, then cardid and checklistid will be set too
+		if areAPIKeysSet() {
+			configData.ConfKey = trelloKey
+			configData.ConfToken = trelloToken
+			configData.ConfCardID = trelloCardID
+			configData.ConfChecklistID = trelloChecklistID
+		}
+		err = CheckUpdates(jsonFile, configData)
+		if err != nil {
+			log.Fatal(err)
+		}
+	} else {
+		if verbose {
+			log.Printf("This is a test update since API keys are not set.\n")
+			err := CheckUpdates(testJSONFile, testConfigData)
+			if err != nil {
+				log.Fatal(err)
+			}
+		}
+	}
+
+}
+
+// areAPIKeysSet checks if api keys and tokens are set
+func areAPIKeysSet() bool {
+	if (trelloKey != "" && trelloKey != "key") || (trelloToken != "" && trelloToken != "token") {
+		return true
+	}
+	return false
+}
+
+// getSha gets the sha of the latest commit
+func getSha(repoPath string) (ShaResponse, error) {
+	var resp ShaResponse
+	path := fmt.Sprintf(githubPath, repoPath)
+	if verbose {
+		fmt.Println(path)
```

Comment:
Be more descriptive here and include that this is getting the SHA from the repository path.